orchagent/debug_counter/drop_counter.cpp

```diff
@@ -342,6 +342,61 @@ unordered_set<string> DropCounter::getSupportedDropReasons(sai_debug_counter_att
     return supported_drop_reasons;
 }
 
+// Returns a set of supported counter types.
+unordered_set<string> DropCounter::getSupportedCounterTypes()
+{
+    sai_status_t status = SAI_STATUS_FAILURE;
+
+    const auto& countersTypeLookup = getDebugCounterTypeLookup();
+    unordered_set<string> supportedCounterTypes;
+
+    sai_s32_list_t enumValuesCapabilities;
+    vector<int32_t> saiCounterTypes;
+
+    const auto* meta = sai_metadata_get_attr_metadata(SAI_OBJECT_TYPE_DEBUG_COUNTER,
+                                                      SAI_DEBUG_COUNTER_ATTR_TYPE);
+    if (!meta)
+    {
+        SWSS_LOG_THROW("SAI BUG: metadata null pointer returned by "
```

Comment:
Please specify ERROR log and return. No need to throw here. Same for below case as well

Reply:
Done